tests/functional/aws-node-sdk/test/object/mpuVersion.js

```diff
@@ -0,0 +1,748 @@
+const assert = require('assert');
+const async = require('async');
+const { versioning } = require('arsenal');
+
+const { config } = require('../../../../../lib/Config');
+const withV4 = require('../support/withV4');
+const BucketUtility = require('../../lib/utility/bucket-util');
+const metadata = require('../../../../../lib/metadata/wrapper');
+const { DummyRequestLogger } = require('../../../../unit/helpers');
+const checkError = require('../../lib/utility/checkError');
+
+const versionIdUtils = versioning.VersionID;
+
+const log = new DummyRequestLogger();
+
+const nonVersionedObjId =
+    versionIdUtils.getInfVid(config.replicationGroupId);
+const bucketName = 'bucket1putversion31';
+const objectName = 'object1putversion';
+const mdListingParams = { listingType: 'DelimiterVersions', maxKeys: 1000 };
+
+function _getMetadata(bucketName, objectName, versionId, cb) {
+    let decodedVersionId;
+    if (versionId) {
+        if (versionId === 'null') {
```

Comment:
```suggestion
        if (versionId === null) {
```

Reply:
`versionId === 'null'` will fail!

Reply:
What do you mean?!

Reply:
Is `'null'` a string returned here? if yes, then it is correct. Otherwise it will fail no?

Reply:
It is expecting a 'null' string. `null` won't have passed the previous `if (versionId) {`

Reply:
Yes and I missed it completely and if that is the case would versionId be null or undefined ever?